lib/block_view_nft.go

```diff
@@ -772,143 +1036,182 @@ func (bav *UtxoView) _connectAcceptNFTBid(
 		// in an NFT collection. Thus, we must check to see if a SerialNumber zero bid exists
 		// for this bidder before we return an error.
-		nftBidKey = MakeNFTBidKey(txMeta.BidderPKID, txMeta.NFTPostHash, uint64(0))
+		nftBidKey = MakeNFTBidKey(args.BidderPKID, args.NFTPostHash, uint64(0))
 		nftBidEntry = bav.GetNFTBidEntryForNFTBidKey(&nftBidKey)
 		if nftBidEntry == nil || nftBidEntry.isDeleted {
-			return 0, 0, nil, RuleErrorCantAcceptNonExistentBid
+			return 0, 0, nil, errors.Wrapf(RuleErrorCantAcceptNonExistentBid, "_helpConnectNFTSold: ")
 		}
 	}
-	if nftBidEntry.BidAmountNanos != txMeta.BidAmountNanos {
-		return 0, 0, nil, RuleErrorAcceptedNFTBidAmountDoesNotMatch
+
+	if nftBidEntry.BidAmountNanos != args.BidAmountNanos {
+		return 0, 0, nil, errors.Wrapf(RuleErrorAcceptedNFTBidAmountDoesNotMatch, "_helpConnectNFTSold: ")
 	}
 
-	bidderPublicKey := bav.GetPublicKeyForPKID(txMeta.BidderPKID)
+	bidderPublicKey := bav.GetPublicKeyForPKID(args.BidderPKID)
 
 	//
 	// Store starting balances of all the participants to check diff later.
 	//
 	// We assume the tip is right before the block in which this txn is about to be applied.
 	tipHeight := uint32(0)
+	blockHeight := args.BlockHeight
 	if blockHeight > 0 {
 		tipHeight = blockHeight - 1
 	}
-	sellerBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(txn.PublicKey, tipHeight)
+	sellerPublicKey := bav.GetPublicKeyForPKID(prevNFTEntry.OwnerPKID)
+	sellerBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(sellerPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for seller pubkey: %v",
-			PkToStringBoth(txn.PublicKey))
+			"_helpConnectNFTSold: Problem getting initial balance for seller pubkey: %v",
+			PkToStringBoth(sellerPublicKey))
 	}
 	bidderBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(
 		bidderPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for bidder pubkey: %v",
+			"_helpConnectNFTSold: Problem getting initial balance for bidder pubkey: %v",
 			PkToStringBoth(bidderPublicKey))
 	}
 	creatorBalanceBefore, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(
 		nftPostEntry.PosterPublicKey, tipHeight)
 	if err != nil {
 		return 0, 0, nil, fmt.Errorf(
-			"_connectAcceptNFTBid: Problem getting initial balance for poster pubkey: %v",
+			"_helpConnectNFTSold: Problem getting initial balance for poster pubkey: %v",
 			PkToStringBoth(nftPostEntry.PosterPublicKey))
 	}
 
-	//
-	// Validate bidder UTXOs.
-	//
-	if len(txMeta.BidderInputs) == 0 {
-		return 0, 0, nil, RuleErrorAcceptedNFTBidMustSpecifyBidderInputs
+	// Connect basic txn to get the total input and the total output without
+	// considering the transaction metadata.
+	totalInput, totalOutput, utxoOpsFromBasicTransfer, err := bav._connectBasicTransfer(
```

Comment:
See Slack for discussion

Reply:
Yeah, I think this needs to move to be *after* we connect the bidderInputs. Right now, the way you have it, I think the utxoOperations are getting ordered funny because you're doing:
```
// Connect basic transfer
// Connect bidder inputs
// Add bidder utxoOps to utxoOperations (this is backwards)
// Add basic transfer utxoOps to utxoOperations
```

But it needs to be:
```
// Connect bidder inputs
// Add bidder utxoOps to utxoOperations (this is backwards)
// Connect basic transfer
// Add basic transfer utxoOps to utxoOperations
```